controllers/session/conditions.go

```diff
@@ -0,0 +1,84 @@
+package session
+
+import (
+	"strconv"
+
+	istiov1alpha1 "github.com/maistra/istio-workspace/api/maistra/v1alpha1"
+	"github.com/maistra/istio-workspace/pkg/model"
+)
+
+func addConditionForLocatedRef(session *istiov1alpha1.Session, ref model.Ref, modified *model.LocatorStatus) {
+	message := modified.Kind + "/" + modified.Name + " located " + ref.KindName.String() + ": "
+	reason := toReason(modified.Action)
+	typeStr := string(modified.Action)
+	located := "true"
+	session.AddCondition(istiov1alpha1.Condition{
+		Source: istiov1alpha1.Source{
+			Kind: modified.Kind,
+			Name: modified.Name,
+			Ref:  ref.KindName.String(),
+		},
+		Message: &message,
+		Reason:  &reason,
+		Status:  &located,
+		Type:    &typeStr,
+	})
+}
+
+func addConditionForModifiedRef(session *istiov1alpha1.Session, ref model.Ref, modified *model.ModificatorStatus) {
```

Comment:
Maybe we should evaluate to make these session.AddCondition(createConditionForModifiedRef)

Reply:
I was actually thinking about doing it differently. The current approach is a bit too procedural at this point. I will change it as it makes more sense.

Reply:
Improved in 4877cf2